### PR TITLE
adding rust

### PR DESCRIPTION
**Pull request by softchris**
installing Rust

### CODE REVIEW

File: linux/base.Dockerfile [hunk 1]
```diff
@@ -202,6 +202,30 @@ RUN wget -nv -O puppet-tools.deb https://apt.puppet.com/puppet-tools-release-bus
...
+  apt-get install -y --no-install-recommends \
+  ca-certificates \
+  gcc \
+  libc6-dev \
+  wget \
+  ; \
+  \ 
+  wget https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init \
+  chmod +x rustup-init; \
+  ./rustup-init -y --no-modify-path --default-toolchain nightly; \
```
Comment (rylev) [inline linux/base.Dockerfile:222]: What’s the reason for defaulting to nightly?

File: linux/base.Dockerfile [hunk 2]
```diff
@@ -202,6 +202,30 @@ RUN wget -nv -O puppet-tools.deb https://apt.puppet.com/puppet-tools-release-bus
...
+RUN RUN set -eux; \
+  apt-get update; \
+  apt-get install -y --no-install-recommends \
+  ca-certificates \
+  gcc \
+  libc6-dev \
+  wget \
+  ; \
+  \ 
+  wget https://static.rust-lang.org/rustup/dist/x86_64-unknown-linux-gnu/rustup-init \
```
Comment (rylev) [inline linux/base.Dockerfile:220]: Are you forgetting a semi-colon at the end of this line?

File: linux/base.Dockerfile [hunk 3]
```diff
@@ -202,6 +202,30 @@ RUN wget -nv -O puppet-tools.deb https://apt.puppet.com/puppet-tools-release-bus
...
+ENV RUSTUP_HOME=/usr/local/rustup \
+  CARGO_HOME=/usr/local/cargo \
+  PATH=/usr/local/cargo/bin:$PATH
+
+RUN RUN set -eux; \
+  apt-get update; \
+  apt-get install -y --no-install-recommends \
+  ca-certificates \
+  gcc \
+  libc6-dev \
```
Comment (rylev) [inline linux/base.Dockerfile:216]: I’m curious why this package is needed. Doesn’t the system come with libc on it?

**Review by rylev — COMMENTED** [review]
Overall, looks good! Just have a few questions.

File: linux/base.Dockerfile (hunk 3 above)
Comment (softchris) [reply linux/base.Dockerfile:216]: was trying to follow this guide, https://hub.docker.com/r/rustlang/rust/dockerfile @rylev

File: linux/base.Dockerfile (hunk 1 above)
Comment (softchris) [reply linux/base.Dockerfile]: @rylev good point. That seems to be what Docker Hub image has but it makes more sense to use the stable version imo

**Comment by dsajanice** [comment]
We don't have plans for adding Rust to our image due to size limitations. We'll monitor requests if they come in and evaluate.

**Pull request closed without merging**